File: Bit/MajorityNumber/MajorityNumber.cc
class Solution {
	public:
		int majorityNumber(vector<int> nums) {
			if(nums.empty())
				return -1;
			
			int key = -1, count = 0;
			for(auto num: nums) {
				if(count == 0) {
					key = num;
					count = 1;
					continue;
				}
				
				if(key == num) {
					count++;
				} else {
					count--;
				}
			}
			return key;
		}
};

File: Bit/MajorityNumber/MajorityNumber.py
class Solution:
    def majorityNumber(self, nums):
        if nums is none:
            return -1
            
        key, count = -1, 0    
        for i in nums:
            if count == 0:
                key = i
                count = 1
                continue
                
            if key == num:
                count += 1
            else:
                count -= 1
                
        return key